### PR TITLE
Add component name to property validation error message

### PR DESCRIPTION
**Pull request by ConorGriffin37**
Add componentName to property validation error message. This would be helpful in cases where element tracing is not turned on or provides an unhelpful error message.

Checklist before submitting:
* [x] Added entry to `CHANGELOG.md`
* [x] Added/updated relevant tests
* [x] Added/updated documentation

### CODE REVIEW

File: src/Component.spec/validateProps.spec.lua [hunk 1]
```diff
@@ -188,7 +188,8 @@ return function()
 			end)
 
 			expect(success).to.equal(false)
-			expect(error:find("MyComponent")).to.be.ok()
+			local startIndex = error:find("MyComponent")
+			expect(startIndex).to.be.ok()
```
Comment (ZoteTheMighty) [inline src/Component.spec/validateProps.spec.lua:192]: Oh, good catch here!

**Comment by ConorGriffin37** [comment]
@cliffchapmanrbx Looks like this CLA check doesn't work with forks, not sure if we need to change it?

I'll just add myself to the whitelist since I work at Roblox I presume this doesn't apply to me.

**Comment by cliffchapmanrbx** [comment]
Minor deployment bugs on the bot that I fixed late yesterday, it should work normally now. As it says at http://go/clabot FTEs should not sign the CLA, they should be whitelisted.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased Changes
 
+* Added component name to property validation error message
```
Comment (LPGhatguy) [inline CHANGELOG.md:5]:
```suggestion
* Added component name to property validation error message ([#275](https://github.com/Roblox/roact/pull/275))
```
Linking back to the PR containing a change is something we forget to do a lot, but is super useful when looking back over these!

**Pull request merged**